src/bindings/c/include/openvino/c/ov_model.h

```diff
@@ -32,44 +32,164 @@ typedef struct ov_model ov_model_t;
 OPENVINO_C_API(void) ov_model_free(ov_model_t* model);
 
 /**
- * @brief Get the outputs of ov_model_t.
+ * @brief Get const output ports list of ov_model_t.
  * @ingroup model
  * @param model A pointer to the ov_model_t.
- * @param output_nodes A pointer to the ov_output_nodes.
+ * @param output_ports A pointer to the ov_output_nodes.
  * @return Status code of the operation: OK(0) for success.
  */
-OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_nodes);
+OPENVINO_C_API(ov_status_e) ov_model_const_outputs(const ov_model_t* model, ov_output_const_node_list_t* output_ports);
 
 /**
- * @brief Get the outputs of ov_model_t.
+ * @brief Get const input ports list of ov_model_t.
  * @ingroup model
  * @param model A pointer to the ov_model_t.
- * @param input_nodes A pointer to the ov_input_nodes.
+ * @param input_ports A pointer to the ov_input_nodes.
  * @return Status code of the operation: OK(0) for success.
  */
-OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_nodes);
+OPENVINO_C_API(ov_status_e) ov_model_const_inputs(const ov_model_t* model, ov_output_const_node_list_t* input_ports);
 
 /**
- * @brief Get the outputs of ov_model_t.
+ * @brief Get output ports list of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param output_ports A pointer to the ov_output_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_outputs(const ov_model_t* model, ov_output_node_list_t* output_ports);
+
+/**
+ * @brief Get input ports list of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param input_ports A pointer to the ov_input_nodes.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_inputs(const ov_model_t* model, ov_output_node_list_t* input_ports);
+
+/**
+ * @brief Get a const input port of ov_model_t.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param input_port A pointer to the ov_output_const_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_const_input(const ov_model_t* model, ov_output_const_node_t** input_port);
```

Comment:
BTW, why do we need a list of inputs or outputs ports? What if we just have API to query number of inputs / outputs and query specific input / output individually? 
In what cases we need a list?

Reply:
I see that with such list `ov_output_node_list_t` we always need to pass index, so why not to query specific node_output from model by index, then take any kind of information from this node_output?

So, to conclude `ov_output_const_node_t` seems not needed for me

Reply:
Actually, such api is called in hello_classification_c demo and we wrap it just to make sample can run.